### PR TITLE
Addon: Added google analytics

### LINKED ISSUE #3944: Google analytics
### Bug or support request summary

Hi I am wondering how to add google analytics on storybook?

I have added the google analytics script to `manager-head.html` so it appears in the `head` which is good.
But because storybook is a one page app google analytics isn't recording the routed url.

### Steps to reproduce

Clicking on sub pages or components in the side bar is not being recorded in google analytics.
This is would be really handy to get some statistics on popular components and track user flows.

### Please specify which version of Storybook and optionally any affected addons that you're running

```
    "@storybook/addon-knobs": "3.4.5",
    "@storybook/addon-options": "3.4.5",
    "@storybook/react": "3.4.5",
    "@storybook/storybook-deployer": "2.3.0",
```

### Affected platforms 

- All platforms

### Screenshots / Screencast / Code Snippets (Optional)

Added in `manager-head.html`

```js
  <!-- Analytics -->
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
    ga('create', 'ID', 'auto');
    ga('send', 'pageview');
  </script>
```

### PR DESCRIPTION
**Pull request by pksunkara**
Fixes #3944 

This PR adds a google analytics addon. This is not ready to be merged. Just a call for review for now.

### TODO

- [ ] Add integration tests
- [x] List the addon in docs

### CODE REVIEW

File: addons/google-analytics/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Then, add following content to `.storybook/addons.js`
+
+```js
+import '@storybook/addon-google-analytics/register';
+```
+
+Then, set an environment variable
+
+```
+STORYBOOK_GA_ID = UA-000000-01
```
Comment (ndelangen) [inline addons/google-analytics/README.md]: Is this a global variable?

Might want to make that explicit:
```
window.STORYBOOK_GA_ID = UA-000000-01;
```

**Comment by ndelangen** [comment]
So I don't think we want this in any example we have.. since really I don't fancy getting into tracking people who browser our examples.

So maybe add some unit test for this?

**Comment by informa** [comment]
Awesome work, is this addon nearly finished?

**Comment by pksunkara** [comment]
Need some docs work and tests.

**Comment by ndelangen** [comment]
I actually think the code is simple enough to not have any tests tbh.

Let's add some documentation and merge, so we can announce this with 4.0.0 @pksunkara

**Comment by pksunkara** [comment]
@ndelangen This PR is ready to go.

**Review by ndelangen — APPROVED** [review]

**Comment by igor-dv** [comment]
It should be added in https://github.com/storybooks/storybook/blob/master/ADDONS_SUPPORT.md
and in https://github.com/storybooks/storybook#addons

**Pull request merged**